cordova-plugin-outline/apple/src/OutlineConnectivity.swift

```diff
@@ -12,36 +12,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import CocoaAsyncSocket
 import CocoaLumberjack
 import CocoaLumberjackSwift
+import Tun2socks
 
 // Class to perform connectivity tests against remote Outline servers.
-class OutlineConnectivity: NSObject, GCDAsyncSocketDelegate {
+class OutlineConnectivity: NSObject {
 
-  private let kSocketTimeoutSecs = 10.0
-
-  private var reachabilityCallbackBySocket = [GCDAsyncSocket: ((Bool) -> Void)]()
-  private let delegateQueue = DispatchQueue(label: "tcp")
-
-  /**
-    Asynchronously calls |completion| with a boolean representing whether |host| and |port|
-    are reachable.
-   */
+  // Asynchronously calls |completion| with a boolean representing whether |host| and |port|
+  // are reachable.
   func isServerReachable(host: String, port: UInt16, completion: @escaping((Bool) -> Void)) {
     DispatchQueue.global(qos: .background).async {
       guard let networkIp = self.getNetworkIpAddress(host) else {
         DDLogError("Failed to retrieve the remote host IP address in the network");
         return completion(false)
       }
-      let socket = GCDAsyncSocket(delegate: self, delegateQueue: self.delegateQueue)
-      self.reachabilityCallbackBySocket[socket] = completion
-      do {
-        try socket.connect(toHost: networkIp, onPort: port, withTimeout: self.kSocketTimeoutSecs)
-      } catch {
-        DDLogError("Unexpected error in reachability connection")
-        self.sendSocketReachability(socket, reachable: false)
-      }
+      var error: NSError?
+      ShadowsocksCheckServerReachable(networkIp, Int(port), &error)
```

Comment:
This returns a `BOOL`, like `ShadowsocksCheckConnectivity`, so you should be able to write this as:

```objc
completion(ShadowsocksCheckServerReachable(networkIp, Int(port), nil))
```

Reply:
Done.